integreat_cms/cms/views/contacts/contact_bulk_actions.py

```diff
@@ -0,0 +1,176 @@
+    """
+
+    #: The model of this :class:`~integreat_cms.cms.views.bulk_action_views.BulkActionView`
+    model = Contact
+
+
+class ArchiveContactBulkAction(
+    ContactBulkActionMixin, ContactBulkAction
+):  # pylint: disable=too-many-ancestors
+    """
+    Bulk action to archive multiple contacts at once
+    """
+
+    def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        r"""
+        Archive multiple contacts at once
+
+        :param request: The current request
+        :param \*args: The supplied arguments
+        :param \**kwargs: The supplied keyword arguments
+        :return: The redirect
+        """
+
+        archive_successful = []
+
+        for content_object in self.get_queryset():
+            content_object.archive()
+            archive_successful.append(content_object)
+
+        if archive_successful:
+            messages.success(
+                request,
+                ngettext_lazy(
+                    "{model_name} {object_names} was successfully archived.",
+                    "The following {model_name_plural} were successfully archived: {object_names}",
+                    len(archive_successful),
+                ).format(
+                    model_name=self.model._meta.verbose_name.title(),
+                    model_name_plural=self.model._meta.verbose_name_plural,
+                    object_names=iter_to_string(archive_successful),
+                ),
+            )
+
+        return super().post(request, *args, **kwargs)
+
+
+class RestoreContactBulkAction(
+    ContactBulkActionMixin, ContactBulkAction
+):  # pylint: disable=too-many-ancestors
+    """
+    Bulk action to restore multiple contacts at once
+    """
+
+    def post(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        r"""
+        Function to restore multiple contacts at once
+
+        :param request: The current request
+        :param \*args: The supplied arguments
+        :param \**kwargs: The supplied keyword arguments
+        :return: The redirect
+        """
+        restore_sucessful = []
+        for content_object in self.get_queryset():
+            content_object.restore()
+            restore_sucessful.append(content_object)
+
+        if restore_sucessful:
+            messages.success(
+                request,
+                ngettext_lazy(
+                    "{model_name} {object_names} was successfully restored.",
+                    "The following {model_name_plural} were successfully restored: {object_names}",
+                    len(restore_sucessful),
+                ).format(
+                    model_name=self.model._meta.verbose_name.title(),
+                    model_name_plural=self.model._meta.verbose_name_plural,
+                    object_names=iter_to_string(restore_sucessful),
+                ),
+            )
+
+        return super().post(request, *args, **kwargs)
+
+
+class DeleteContactBulkAction(
```

Comment:
I know it was part of the issue, but this would be our first delete bulk action afaik. I think if we include the delete action here, we should also consider implementing it for the other bulk actions. @osmers Do you have an opinion on that?

Reply:
I think so too. It's at least not part of `bulk_actions_view.py`. It's also part of issue #2950